x/ethbridge/keeper/firstLockDoublePeg.go

```diff
@@ -0,0 +1,48 @@
+package keeper
+
+import (
+	"strconv"
+
+	"github.com/Sifchain/sifnode/x/ethbridge/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetFirstLockDoublePeg get first lock double peg
+func (k Keeper) GetFirstLockDoublePeg(ctx sdk.Context, denom string) bool {
```

Comment:
Why are we creating a new keeper for this? Shouldn't the token registry be keeping track of deployed tokens on cross chains? If so can we just query the token registry to see if this is our first double peg?

Reply:
I just realize I forget other parameter as NetworkDescriptor. and I prefer separate storage for it.

Reply:
So I guess the reason I would think this should be part of the token registry is two-part. 

1. I would imagine the token registry as a snapshot of the current state of deployed tokens through sifchain so from a single query I can determine token metadata and deployed contracts across chains along with if tokens are IBC native or not. Rather then having to piece information together from many sources.
2. We have to determine not only if a token has been deployed to a chain before but we also need to know if a token is native to a chain or if its a non-native asset (IBC or double pegged) to ensure that double pegged assets are burned and not locked, since we already need a mechanism to track double pegged assets from native assets and this information would need to be quired from multiple EVM chains it would make sense then if there exists no record of the asset in the token registry when you go to peg it out to an EVM chain that its A) a nonnative asset and B) that its a first-time peg of a non-native asset.

Does this make sense? If anyone else wants to chime in please do, I would just think having a central store for the state of token metadata would make sense. And I would think this information could be gleaned from redundant information that needs to exist to determine if a token is EVM native or not.